pkg/datapath/linux/probes/probes_test.go

```diff
@@ -344,3 +344,16 @@ func TestHaveTCX(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestDetectHaveFibIfindex(t *testing.T) {
+	testutils.PrivilegedTest(t)
+	testutils.SkipOnOldKernel(t, "5.10", "bpf: Always return target ifindex in bpf_fib_lookup")
```

Comment:
Humm.. `HaveFibIfindex` call will return `err` = `ebpf.ErrNotSupported`, so no, the test - as written here - shouldn't succeed for Kernels that don't have the patch.

But as I say in the comment, I see little value on this unit, as you don't have much control on which Kernel the CI is actually running. I think a much better approach would be to run it as part of the e2e or integr, where tests are run against specific kernels, making sure detection (true/false) is correct based on the specific kernel version.

Reply:
Good question, I don't actually know how we test these. ci-runtime only runs on whatever CI has.

Reply:
@julianwiedmann?

Reply:
@lmb I was discussing with Marc a bit more. Should this probe actually live in `cilium/ebpf/features`, and would `cilium/ebpf` have infrastructure for more detailed testing of the probe code, across different kernels etc?

Reply:
Yes, testing of the probe would definitely be more thorough in c/ebpf since we have CI against all supported kernel.org releases. Pulling in @rgo3 since he maintains `features`.

Reply:
I think the current split of having `c/ebpf/features` and `pkg/datapath/linux/probes` is intentional. `c/ebpf/features` was only intended for more generalized APIs, like the ones we currently have, e.g. `HaveProgramType`, `HaveMapType`,  `HaveProgramHelper`. Specifically with `HaveProgramHelper` there is the issue that it only cares about the presence of a certain helper, callers can't really know which specific functionality this helper offers if it's behavior changed over time in the kernel. IMO if individual projects (including Cilium) need to know about specific behavior those projects should maintain more specific probes themselves. So I think this probe is in the right place, but we might need to improve Cilium's CI to run `ci-runtime` tests against different kernel versions if otherwise the testing of our internal probes doesn't have much value.

Reply:
To add, package features was initially designed to mirror bpftool. Adding this probe there seems a little arbitrary (makingi it hard to draw a line). If the agent needs more bpftool probes, we should put those into the library.